File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix = strs[0];
        if(strs.length==0) return "";
        for(int i=0;i<strs.length;i++){
            while(strs[i].indexOf(prefix)!=0){
                prefix = prefix.substring(0,prefix.length()-1);
                if(prefix.isEmpty()) return  "";
            }
        }

        return prefix;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        int left=0;
        string ans="";
        string first=strs[0];
        string end=strs[strs.size()-1];
        for(int i=0;i<min(first.size(),end.size());i++){
            if(first[i]!=end[i]){
                return ans;
            }
            ans+=first[i];
        }
        return ans;

        
    }
};